syntax = "proto3";
package parking;

service ParkingDetector {
  rpc Predict(PredictRequest) returns (PredictResponse) {}
}

message PredictRequest {
  bytes image = 1;   // JPEG/PNG bytes
  float conf  = 2;   // 0..1
  int32 imgsz = 3;   // ej. 640
}

message Box {
  float x1 = 1;
  float y1 = 2;
  float x2 = 3;
  float y2 = 4;
  int32 cls = 5;
  float conf = 6;
  string name = 7;
}

message PredictResponse {
  repeated Box boxes     = 1;
  int32 free_count       = 2;
  int32 occupied_count   = 3;
  bytes annotated        = 4; // JPG con overlays
}